src/ansi-c/type2name.cpp

```diff
@@ -168,7 +177,7 @@ static std::string type2name(
     const array_typet &t=to_array_type(type);
     mp_integer size;
     if(t.size().id()==ID_symbol)
-      result+="ARR"+t.size().get_string(ID_identifier);
+      result+="ARR"+id2string(t.size().id());
```

Comment:
No, that would yield "symbol". Use `id2string(t.size().get(ID_identifier))`

Reply:
done